src/main/scala/com/typesafe/sbt/packager/jar/ClasspathJarPlugin.scala

```diff
@@ -1,46 +1,54 @@
+import com.typesafe.sbt.SbtNativePackager.Universal
 
 object ClasspathJarPlugin extends AutoPlugin {
 
   object autoImport {
-    val classspathJarName = TaskKey[String]("classpath-jar-name", "classpath-jar name")
+    val packageJavaClasspathJar = TaskKey[File]("packageJavaClasspathJar", "Creates a Java classpath jar that specifies the classpath in its manifest")
   }
   import autoImport._
 
   override def requires = archetypes.JavaAppPackaging
 
-  override lazy val projectSettings = Seq[Setting[_]](
-    classspathJarName <<= (Keys.packageBin in Compile, Keys.projectID, Keys.artifact in Compile in Keys.packageBin) map { (jar, id, art) =>
-      makeJarName(id.organization, id.name, id.revision, art.name, art.classifier)
+  override lazy val projectSettings = Defaults.packageTaskSettings(packageJavaClasspathJar, mappings in packageJavaClasspathJar) ++ Seq(
+    mappings in packageJavaClasspathJar := Nil,
+
+    artifactClassifier in packageJavaClasspathJar := Option("classpath"),
+
+    packageOptions in packageJavaClasspathJar := {
+      val classpath = (scriptClasspath in packageJavaClasspathJar).value
+      val manifestClasspath = Attributes.Name.CLASS_PATH -> classpath.mkString(" ")
+      Seq(
+        ManifestAttributes(manifestClasspath)
+      )
     },
-    mappings in Universal += {
-      ((target in Universal).value / "lib" / classspathJarName.value) -> ("lib/" + classspathJarName.value)
+
+    artifactName in packageJavaClasspathJar := { (scalaVersion, moduleId, artifact) =>
+      moduleId.organization + "." + artifact.name + "-" + moduleId.revision +
+        artifact.classifier.fold("")("-" + _) + "." + artifact.extension
     },
-    scriptClasspath := {
-      writeJar((target in Universal).value / "lib" / classspathJarName.value, scriptClasspath.value)
-      Seq(classspathJarName.value)
+
+    scriptClasspath in bashScriptDefines := Seq(
+      (artifactPath in packageJavaClasspathJar).value.getName
+    ),
+
+    scriptClasspath in batScriptReplacements := Seq(
+      (artifactPath in packageJavaClasspathJar).value.getName
+    ),
+
+    mappings in Universal += {
+      val classpathJar = packageJavaClasspathJar.value
+      classpathJar -> ("lib/" + classpathJar.getName)
     }
-  )
 
-  // Constructs a jar name from components...(ModuleID/Artifact)
-  private def makeJarName(org: String, name: String, revision: String, artifactName: String, artifactClassifier: Option[String]): String =
-    (org + "." +
-      name + "-" +
-      Option(artifactName.replace(name, "")).filterNot(_.isEmpty).map(_ + "-").getOrElse("") +
-      revision +
-      artifactClassifier.filterNot(_.isEmpty).map("-" + _).getOrElse("") +
-      "-classpath.jar")
-
-  /** write jar file */
-  private def writeJar(jarFile: File, allClasspath: Seq[String]) = {
-    val manifest = new java.util.jar.Manifest()
-    manifest.getMainAttributes().putValue("Class-Path", allClasspath.mkString(" "))
-    IO.jar(Seq.empty, jarFile, manifest)
-  }
```

Comment:
No reason to write own own code for building a jar when sbt already has a very powerful and flexible jar building mechanism.